src/pybind/mgr/k8sevents/module.py

```diff
@@ -0,0 +1,1029 @@
+        {
+            "cmd": "k8sevents namespace",
+            "desc": "Show all Kuberenetes events from the rook-ceph namespace",
+            "perm": "r"
+        },
+        {
+            "cmd": "k8sevents ceph",
+            "desc": "Show only Ceph related events tracked & sent to the kubernetes cluster",
+            "perm": "r"
+        }
+    ]
+    MODULE_OPTIONS = [
+        {'name': 'log_level'},
+        {'name': 'heartbeat_interval_secs'},
+        {'name': 'config_check_secs'}
+    ]
+
+    def __init__(self, *args, **kwargs):
+        self.run = True
+        self.event_processor = None
+        self.config_watcher = None
+        self.ns_watcher = None
+        super(Module, self).__init__(*args, **kwargs)
+
+    def fetch_events(self):
+        """Interface to expose current events to another mgr module"""
+        # TODO
+        return dict()
+
+    def show_all_events(self):
+        """Show all events we're holding from the ceph namespace - most recent 1st"""
+
+        max_name_length = max([len(k) for k in self.ns_watcher.events])
+        max_msg_length = max([len(self.ns_watcher.events[k].message) for k in self.ns_watcher.events])
+        fmt = "{:<20}  {:>5}  {:<" + str(max_msg_length) + "}  {:<" + str(max_name_length) + "}\n"
+        s = fmt.format("Last Seen", "Count", "Message", "Event Object Name")
+
+        for event_name in sorted(self.ns_watcher.events, 
+                                 key = lambda name: self.ns_watcher.events[name].last_timestamp,
+                                 reverse=True):
+
+            event = self.ns_watcher.events[event_name]
+
+            s += fmt.format(
+                    datetime.strftime(event.last_timestamp,"%Y/%m/%d %H:%M:%S"),
+                    event.count,
+                    event.message,
+                    event_name
+            )
+
+        return 0, "", s
+
+    def show_ceph_events(self):
+        if len(self.event_processor.events.keys()) > 0:
+            s = "\n".join(self.event_processor.events.keys())
+        else:
+            s = "No events\n"
+        return 0, "", s
+
+    def handle_command(self, inbuf, cmd):
+        # TODO Should we implement dynamic options for the monitoring?
+        if cmd["prefix"] == "k8sevents namespace":
+            return self.show_all_events()
+        elif cmd["prefix"] == "k8sevents ceph":
+            return self.show_ceph_events()
+        else:
+            raise NotImplementedError(cmd["prefix"])
+
+    @staticmethod
+    def can_run():
+        """Determine whether the pre-reqs for the module are in place"""
+
+        if not kubernetes_imported:
+            return False, "kubernetes python client is unavailable"
+        return True, ""
+
+    def log_callback(self, arg, line, channel, name, who, stamp_sec, stamp_nsec, seq, level, msg):
+        """Receive the ceph log entry and add it to the global queue"""
+
+        if sys.version_info[0] >= 3:
+            channel = channel.decode('utf-8')
```

Comment:
@neha-ojha why do we need to make this conversion in Python 3? Wouldn't it make sense to also just make it a string?